### DISCUSSION TITLE
Getting java.lang.NoSuchMethodError when running new v4.1.2 install

### DISCUSSION THREAD

**Discussion by markgif** [dita-ot/dita-ot#4378] [Q&A] [answered]
I get this weird error when running a freshly-installed DITA OT 4.1.2:

```
bin\dita -i C:\dita-api-manual\sample.ditamap -f pdf2 --temp=temp -v
Searching for integrator.xml ...
init:
dita2pdf2.init:
init-properties:
[init-project] Initializing project
Error: java.lang.NoSuchMethodError: 'net.sf.saxon.Configuration net.sf.saxon.Configuration.newConfiguration()'
```

I'm using Windows 11, using Java 17, openjdk version 17.0.8.1.

My DITA-OT\lib folder contains these files:

C:\dita-ot-4.1.2\lib>dir /b
ant-apache-resolver-1.10.13.jar
ant-launcher.jar
ant.jar
commons-codec-1.15.jar
commons-io-2.8.0.jar
dost-configuration.jar
dost.jar
guava-32.1.1-jre.jar
httpclient5-5.1.3.jar
httpcore5-5.1.3.jar
httpcore5-h2-5.1.3.jar
icu4j-70.1.jar
isorelax-20030108.jar
jackson-annotations-2.15.2.jar
jackson-core-2.15.2.jar
jackson-databind-2.15.2.jar
jackson-dataformat-yaml-2.15.2.jar
jing-20181222.jar
logback-classic-1.4.8.jar
logback-core-1.4.8.jar
Saxon-HE-12.3.jar
slf4j-api-2.0.7.jar
snakeyaml-2.0.jar
xercesImpl-2.12.2.jar
xml-apis-1.4.01.jar
xml-resolver-1.2.jar
xmlresolver-5.2.1-data.jar
xmlresolver-5.2.1.jar

Thanks,
Mark Giffin

**Comment by markgif** [comment]
I fixed this myself. I had forgotten I had installed 2 very old and rather non-standard plugins to the clean DITA OT 4.1.2. I started from scratch again without those 2 plugins, and it ran. Sorry!

But it was an interesting error, wasn't it?

**Reply by raducoravu** [reply]
Maybe one of those plugins had a Saxon library inside it as well?
Initially I suspected your problem might be related to specifying in the system classpath extra JAR libraries like this situation: https://github.com/dita-ot/dita-ot/issues/4342#issuecomment-1851281334

**Reply by markgif** [reply, marked as answer]
Hi Radu, I was too quick to mark this as answered, it was still failing in Win11 command prompt (without my old plugins), even though it worked slightly better in Git Bash.

BUT: 
Your tip about the classpath reminded me that I had set the classpath on this Win11 machine to an old Saxon version for some command line Saxon work, and when I cleared that classpath, DITA OT 4.1.2 worked just fine! It's fixed! Basic classpath user error.

Thanks,
Mark
